src/generators/operationGenerator.ts

```diff
@@ -286,23 +291,44 @@ type ParameterWithDescription = OptionalKind<
 >;
 
 /**
- * Write operations implementation, extracted from OperationGroupDetails, to the generated file
+ * Gets a list of parameter declarations for each overload the operation supports,
+ * and the list of parameter declarations for the base operation.
  */
-export function writeOperations(
-  operationGroupDetails: OperationGroupDetails,
-  operationGroupClass: ClassDeclaration,
-  importedModels: Set<string>,
+function getOperationParameterSignatures(
+  operation: OperationDetails,
   parameters: ParameterDetails[],
-  isInline = false
+  importedModels: Set<string>,
+  operationGroupClass: ClassDeclaration
 ) {
-  operationGroupDetails.operations.forEach(operation => {
-    const responseName = getResponseType(operation, importedModels);
-    const paramDeclarations = filterOperationParameters(
-      parameters,
-      operation
-    ).map<ParameterWithDescription>(param => {
+  const operationParameters = filterOperationParameters(parameters, operation, {
+    includeContentType: true
+  });
+
+  const operationRequests = operation.requests;
+  const overloadParameterDeclarations: ParameterWithDescription[][] = [];
+  const hasMultipleOverloads = Boolean(operationRequests.length > 1);
```

Comment:
What does `Boolean()` do?  Coerce to boolean somehow?

Reply:
Yes! But I don't need it and will remove.

Reply:
Yep, functionally equivalent to `!!` but more clear IMO. Both call the `ToBoolean` abstract operation.